File: ProgJunior/FI.WebAtividadeEntrevista/FI.WebAtividadeEntrevista/Scripts/validaCPF.js
function ValidaCPF() {

    var strCPF = document.getElementById("CPF").value.replace(/[^\w\s]/gi, '');
    var Soma;
    var Resto;

    if (strCPF == "") {
        console.log("CPF vazio.");
        MessageAlert(false);
        return false;
    }

    function allEqual(input) {
        return input.split('').every(char => char === input[0]);
    }

    if (allEqual(strCPF) == true) {
        console.log("CPF com dígitos iguais.");
        MessageAlert(false);
        return false;
    }

    Soma = 0;

    for (i = 1; i <= 9; i++) Soma = Soma + parseInt(strCPF.substring(i - 1, i)) * (11 - i);
    Resto = (Soma * 10) % 11;

    if ((Resto == 10) || (Resto == 11)) Resto = 0;
    if (Resto != parseInt(strCPF.substring(9, 10))) {
        console.log("Primeiro Dígito verificador inválido.");
        MessageAlert(false);
        return false;
    }

    Soma = 0;

    for (i = 1; i <= 10; i++) Soma = Soma + parseInt(strCPF.substring(i - 1, i)) * (12 - i);
    Resto = (Soma * 10) % 11;

    if ((Resto == 10) || (Resto == 11)) Resto = 0;
    if (Resto != parseInt(strCPF.substring(10, 11))) {
        console.log("Segundo Dígito verificador inválido.");
        MessageAlert(false);
        return false;
    }

    document.getElementById("errorCPF").innerHTML = "";
    return true;
}

function MessageAlert(value) {
    if (value == false) {
        document.getElementById("errorCPF").innerHTML = "Digite um CPF válido.";
        document.getElementById("CPF").focus();
    }
}

File: ProgJunior/FI.WebAtividadeEntrevista/FI.WebAtividadeEntrevista/Validate/ValidaCPF.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;
using WebAtividadeEntrevista.Models;
using System.Text.RegularExpressions;

namespace WebAtividadeEntrevista.Validate
{
    public class ValidaCPF : ValidationAttribute
    {
        public string CPF { get; }
        public ValidaCPF()
        {
            
        }

        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
        {
            ValidationResult ErrorMessage = new ValidationResult(this.FormatErrorMessage(validationContext.DisplayName));

            if (value == null)
                return ErrorMessage;
            
            string cpf = Regex.Replace(value.ToString(), @"[\D]", "");
            int soma;
            int resto;

            // Verifica se todos os dígitos são iguais
            if (cpf.All(c => c.Equals(cpf.First())))
                return ErrorMessage;

            // Verifica se CPF é vazio
            if (cpf == "")
                return ErrorMessage;

            // Valida primeiro dígito verificador
            soma = 0;

            for (int i = 1; i <= 9; i++)
                soma += int.Parse(cpf.Substring(i - 1, 1)) * (11 - i);

            resto = (soma * 10) % 11;

            if ((resto == 10) || (resto == 11)) 
                resto = 0;

            if (resto != int.Parse(cpf.Substring(9, 1)))
                return ErrorMessage;

            // Valida segundo dígito verificador
            soma = 0;

            for (int i = 1; i <= 10; i++)
                soma += int.Parse(cpf.Substring(i - 1, 1)) * (12 - i);

            resto = (soma * 10) % 11;

            if ((resto == 10) || (resto == 11))
                resto = 0;

            if (resto != int.Parse(cpf.Substring(10, 1)))
                return ErrorMessage;

            return null;
        }
    }
}